spec/core/ics-004-channel-and-packet-semantics/UPGRADES.md

```diff
@@ -492,8 +503,13 @@ function chanUpgradeAck(
     currentChannel = provableStore.get(channelPath(portIdentifier, channelIdentifier))
     abortTransactionUnless(currentChannel.state == INITUPGRADE || currentChannel.state == TRYUPGRADE)
 
-    // get underlying connection for proof verification
-    connection = getConnection(currentChannel.connectionIdentifier)
+    // connectionHops can change in a channelUpgrade, however both sides must still be each other's counterparty.
+    proposedConnection = provableStore.get(connectionPath(proposedUpgradeChannel.connectionHops[0])
+    abortTransactionUnless(counterpartyChannel.connectionHops[0] == proposedConnection.counterpartyConnectionIdentifier)
+
+    // get underlying connection from the original channel for proof verification
+    originalChannel = provableStore.get(channelRestorePath(portIdentifier, channelIdentifier))
```

Comment:
this will need to change with the restore removal (construct [expected channel end](https://github.com/cosmos/ibc-go/blob/2ed42afa16b2a8f2751cd866bdec9b48f0f32b63/modules/core/04-channel/keeper/upgrade.go#L100))